src/main/java/ml/shifu/shifu/core/dtrain/wdl/optimization/AdaGrad.java

```diff
@@ -31,8 +31,12 @@ public AdaGrad(double learningRate) {
     }
 
     @Override
-    public void update(float[] weight, float[] grad) {
-        if(weight == null || weight.length == 0 || grad == null || grad.length == 0 || weight.length != grad.length) {
+    public void setTrainSize(float trainSize) {
+    }
+
+    @Override
+    public void update(float[] weight, float[] grad, String uniqueKey) {
+        if(weight == null || grad == null || grad.length == 0 || weight.length != grad.length) {
             return;
```

Comment:
exception can be thrown like IllegalStateException because such senario we can get stack trace